smithyctl/internal/command/workflow/parse.go

```diff
@@ -0,0 +1,368 @@
+package workflow
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/distribution/reference"
+	"github.com/go-errors/errors"
+	"gopkg.in/yaml.v3"
+
+	v1 "github.com/smithy-security/smithy/pkg/types/v1"
+
+	"github.com/smithy-security/smithyctl/internal/registry"
+)
+
+type (
+	workflowAdapter struct {
```

Comment:
Allows us to decouple from the v1 type so that we can take decisions around remote/local components fetching